libraries/chain/include/steem/chain/smt_objects/nai_pool_object.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <steem/chain/steem_object_types.hpp>
+#include <steem/protocol/asset_symbol.hpp>
+
+#ifdef STEEM_ENABLE_SMT
+
+namespace steem { namespace chain {
+
+   class nai_pool_object : public object< nai_pool_object_type, nai_pool_object >
```

Comment:
I am interested in your reasoning behind making each NAI in the pool a separate object rather than a singleton object that contains a fixed size array.

Reply:
Changed this to be a singleton, similar to witness schedule.